cdc/sinkv2/ddlsink/cloudstorage/cloud_storage_ddl_sink.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudstorage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/br/pkg/storage"
+	"github.com/pingcap/tiflow/cdc/contextutil"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/sinkv2/ddlsink"
+	"github.com/pingcap/tiflow/cdc/sinkv2/metrics"
+	"github.com/pingcap/tiflow/pkg/sink"
+)
+
+// Assert DDLEventSink implementation
+var _ ddlsink.DDLEventSink = (*ddlSink)(nil)
+
+type ddlSink struct {
+	// id indicates which changefeed this sink belongs to.
+	id model.ChangeFeedID
+	// statistic is used to record the DDL metrics
+	statistics *metrics.Statistics
+	storage    storage.ExternalStorage
+	tableSet   map[int64]struct{}
+}
+
+// NewCloudStorageDDLSink creates a ddl sink for cloud storage.
+func NewCloudStorageDDLSink(ctx context.Context, sinkURI *url.URL) (*ddlSink, error) {
+	// parse backend storage from sinkURI
+	bs, err := storage.ParseBackend(sinkURI.String(), &storage.BackendOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// create an external storage.
+	storage, err := storage.New(ctx, bs, nil)
+	if err != nil {
+		return nil, err
+	}
+	changefeedID := contextutil.ChangefeedIDFromCtx(ctx)
+
+	d := &ddlSink{
+		id:         changefeedID,
+		storage:    storage,
+		tableSet:   make(map[int64]struct{}),
```

Comment:
there is a `tableSet` implemention `https://github.com/pingcap/tiflow/blob/master/cdc/scheduler/internal/v3/scheduler/scheduler_rebalance.go#L229`, if this is reusable, you can move it to the `pkg`.